File: mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);

int main(void)
{
    // Pedindo altura, entre 4 e 8 e maior que 0
    int height = get_height();

    //
}

// Obtendo e válidando altura
int get_height(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while(n < 4 || n > 8);
    return n;
}

File: python/sentimental-mario-more/mario.py
from cs50 import get_int

while True:
    # Pedindo valor ao usuário
    height = get_int("Height: ")
    # Verificando condição
    while height < 1 or height > 8:
        print("Digite um número entre 1 e 8")
        height = get_int("Height: ")
    # Condição verdadeira, quebra o loop
    if height > 0 or height < 9:
        break

for linhas in range(height):
    # Fórmula para os espaços
    # ex: altura = 5, linha - 0, - 1 = 4 espaços
    for espacos in range((height - linhas - 1)):
        print(" ", end="")
    for colunas in range(linhas + 1):
        print("#", end="")

    # Fazer loop da escada para direita
    # espaço do centro
    print("  ", end="")
    for colunas in range(linhas + 1):
        print("#", end="")
    print()
